execution/process.go

```diff
@@ -223,6 +223,11 @@ func (p *ProcessByKey) ReadyForMore(ctx context.Context, tx storage.StateTransac
 	return nil
 }
 
-func (p *ProcessByKey) Close() error {
-	return nil // TODO: Close this, remove state
+func (p *ProcessByKey) Close(ctx context.Context) error {
+	storage := storage.GetStateTransactionFromContext(ctx).GetUnderlyingStorage()
+	if err := storage.DropAll(outputWatermarkPrefix); err != nil {
```

Comment:
This is actually prefixed by the stream ID. (the Next() and GetWatermark() methods get a tx prefixed by the StreamID)

Reply:
So the cleanup in the Pull engine is sufficient.